src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3531,6 +3519,177 @@ private bool NotesAlreadyLoaded(IEnumerable<ExtraAnnotationViewInfo> notes)
         }
         #endregion
 
+        #region Insert Private Methods
+
+        /// <summary>
+        /// Update inserted workspace elements GUIDs 
+        /// </summary>
+        /// <param name="viewInfo">viewInfo</param>
+        /// <param name="ws">workspace</param>
+        private void UpdateWorkspaceGUIDs(ref ExtraWorkspaceViewInfo viewInfo, ref WorkspaceModel ws)
+        {
+            var nodes = ws.Nodes;
+            var connectors = ws.Connectors;
+            var notes = viewInfo?.Annotations;
+
+            var nodesGuids = UpdateNodeGUIDs(ref nodes);
+            var connectorsGuids = UpdateConnectorGUIDs(ref connectors);
+            var notesGuids = UpdateNoteGUIDs(nodesGuids, ref notes);
+
+            // Now propagate these changes to the viewInfo object
+            if (viewInfo == null) return;
+
+            PropagateUpdatesToViewInfo(viewInfo, nodesGuids, notesGuids, connectorsGuids);
+        }
+
+        private Dictionary<string, string> UpdateNoteGUIDs(Dictionary<Guid, Guid> nodesGuids, ref IEnumerable<ExtraAnnotationViewInfo> notes)
+        {
+            Dictionary<string, string> notesGuids = new Dictionary<string, string>();
+
+            // Assign new annotation models new GUIDs
+            // We need two separate runs to first create the new GUIDs and second work with the new GUIDS
+            if (notes != null)
```

Comment:
I find this entire approach a bit complicated...

I am just curious what you think of an approach that modifies the json. I would usually shy away from that approach but you could select all IDS easily BEFORE deserialization or as jtokens and transform them...

I think it would be much less code.

Reply:
Yes! Great call and while I didn't get the idea immediately, I tested it today and it works great and simplifies this whole thing. [Committed here](https://github.com/DynamoDS/Dynamo/pull/13751/commits/d4cc43095260d490fe875a402be0c4f4d693e9c9).

Reply:
Let me know if that doesn't look good to you and I will rework as needed.

Reply:
@mjkkirschner Previously I feel the approach was OK also because we have another task to update Guids while Dynamo users save the current workspace as a new one, then a JSON block of the graph is not guaranteed to be there for a simplified approach. But after second thought, if this approach is preferred, maybe we can do the same for save as  - touch the JSON instead of in-memory workspace representation.